### ISSUE TITLE
run-time error when using hdf5_md

### ISSUE THREAD

**Issue by wkliao** [Parallel-NetCDF/E3SM-IO#75]
When running command-line option `-k -o out.nc -a hdf5_md`,
the following error appears.
```
Error at line 1029 in e3sm_io_driver_hdf5.cpp:
HDF5-DIAG: Error detected in HDF5 (1.11.0) MPI-process 0:
  #000: /hdf5/src/H5Shyper.c line 6889 in H5Sselect_hyperslab(): not a data space
    major: Invalid arguments to routine
    minor: Inappropriate type
Error in var_io_F_case.cpp line 1144 function run_varn_F_case
```

**Comment by wkliao** [comment]
Fix a bug in 29e2469e4de735625393e83e767f0dfc3cfbbdc5
But getting a different error:
```
% src/e3sm_io datasets/f_case_866x72_16p.nc -k -o F_out.h5 -a hdf5_md

  #002: hdf5/src/H5Dio.c line 1183 in H5D__write(): src and dest data space have different sizes
    major: Invalid arguments to routine
    minor: Bad value
```

**Comment by wkliao** [comment]
Use of HDF5 metadata APIs is implemented in PR #152
